### PR TITLE
refactor: consistent identity blocks - data factory

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Works through https://github.com/hashicorp/terraform-provider-azurerm/issues/15187 for Data Factory

Since there's an assumption that the existing `identity_ids` is a list but is now a set, in 3.0 we'll have to make a breaking change to add a field for `customer_managed_key_identity_id` to specify the identity id which should be used for the customer managed key, rather than assuming the first of the `identity_ids`

### CODE REVIEW

**Comment by tombuildsstuff** [comment]
Tests pass:

Screenshot 2022-02-09 at 18 43 35

File: internal/services/datafactory/data_factory_resource_test.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/hashicorp/terraform-provider-azurerm/internal/features"
+
```
Comment (mbfrahry) [inline internal/services/datafactory/data_factory_resource_test.go:9]: We can collapse this
```suggestion
```

File: internal/services/datafactory/data_factory_resource.go [hunk 2]
```diff
@@ -297,14 +308,20 @@ func resourceDataFactoryCreateUpdate(d *pluginsdk.ResourceData, meta interface{}
 			return fmt.Errorf("could not parse Key Vault Key ID: %+v", err)
 		}
 
-		identityIdsRaw := d.Get("identity.0.identity_ids").([]interface{})
+		userAssignedIdentityId := ""
+		if features.ThreePointOhBeta() {
+			userAssignedIdentityId = d.Get("customer_managed_key_identity_id").(string)
+		} else {
+			identityIdsRaw := d.Get("identity.0.identity_ids").([]interface{})
+			userAssignedIdentityId = identityIdsRaw[0].(string)
```
Comment (mbfrahry) [inline internal/services/datafactory/data_factory_resource.go:316]: Should we confirm that `[0]` is there before casting to it?

**Review by mbfrahry — APPROVED** [review]
LGTM with a couple minor comments

File: internal/services/datafactory/data_factory_resource.go (hunk 2 above)
Comment (tombuildsstuff) [reply internal/services/datafactory/data_factory_resource.go:316]: as discussed offline - no this is the existing behaviour and it's guaranteed in the schema, but removed in 3.0

**Pull request merged**